### PR TITLE
Added support for actions outside the chatbot widget

### PR DESCRIPTION
**Pull request by TriangleJuice**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,55 @@ In the index.html, include the core branding stylesheet:
 #### **avatar**
 `string` Avatar image URL. Default is the special provided Antwerp icon.
 
+### Events
+
+#### **actionClicked**
+Triggers when the user clicks a button of a message with type `action`. This type of message can only be generated by the BFF package. The `actionClicked` output event requires a defined action and pauses the workings of the chatbot widget, waiting for a method call (`completeAction`) that has the same defined action. It allows the developer to handle some things outside the chatbot (e.g. a payment) before returning to the chatbot conversation.
```
Comment (vademo) [inline README.md]: This type of message is injected by the BFF by injecting it into the response. The package itself doesn't have the ability to modify a response.

File: README.md [hunk 2]
```diff
@@ -83,6 +83,55 @@ In the index.html, include the core branding stylesheet:
...
+  [url]="'http://localhost:3000/api/chatbot'"
+  [session]="session1"
+  (actionClicked)="triggerMe($event)">
+</aui-chatbot>
+```
+
+1. Your BFF sends a message of type `action` with one or more actions defined.
+
+  ```
+  {
```
Comment (vademo) [inline README.md]: The format of a message is always the same this is a snippet from the response is this what you are documenting?
```js

{
  "type":"action",
  "message":"message to show in button",
  "action":"foo",
  "url":null,
  "image":null
}
```

File: example/app/app.component.ts [hunk 3]
```diff
@@ -7,8 +14,26 @@ import { Component, ViewEncapsulation, ViewChild } from '@angular/core';
   encapsulation: ViewEncapsulation.None,
 })
 export class AppComponent {
+  @ViewChild('myExampleChatbot') myExampleChatbot: ChatbotComponent;
+
   // start new random sessions every time
   public session1: string = Math.random().toString(36).substring(7);
   public session2: string = Math.random().toString(36).substring(7);
```
Comment (vademo) [inline example/app/app.component.ts]: Question unrelated to this pr but why create 2 sessions?

File: README.md (hunk 1 above)
Comment (jsebrech) [reply README.md]: Should this be "actionStarted"? It matches up with "completeAction", and we may at some point trigger actions from something else besides a button.

File: src/chatbot/chatbot.component.ts [hunk 4]
```diff
@@ -51,6 +55,7 @@ export class ChatbotComponent implements OnInit {
   public isLoading = false;
   public loadingIndex: number;
   public isOpen = false;
+  public disableChatbotByAction = '';
```
Comment (jsebrech) [inline src/chatbot/chatbot.component.ts]: "currentAction"? It implies an on-going action. Disabling the UI is a side-effect of there being such an on-going action. In the future there may be other side-effects.

File: README.md (hunk 1 above)
Comment (TriangleJuice) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (TriangleJuice) [reply README.md]: Fixed in the way we discussed.

File: example/app/app.component.ts (hunk 3 above)
Comment (TriangleJuice) [reply example/app/app.component.ts]: Removed redundant code.

File: src/chatbot/chatbot.component.ts (hunk 4 above)
Comment (TriangleJuice) [reply src/chatbot/chatbot.component.ts]: Fixed.

File: README.md (hunk 2 above)
Comment (vademo) [reply README.md]:
```js
    actions: [
        {
            text: "test for button",
            action: "foo"
            params: { foo: "bar" } // optional
        },
    ]
```

**Pull request merged**